### PR TITLE
Add .bash_aliases to shell language

### PR DESCRIPTION
**Pull request by wesinator**
## Description

Add `bash_aliases` to Shell highlighted filenames

## Checklist:

- [x] **I am associating a language with a new file filename.**
  - [x] The new filename is used in hundreds of repositories on GitHub.com
      - [~16,198 code results](https://github.com/search?q=filename%3Abash_aliases+-language%3Ashell+NOT+nothack&type=Code)
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/hivetech/dna/blob/50ad00031be29765b2576fa407d35a36e0608de9/shell/bash/bash_aliases
    - Sample license(s): [Apache](https://github.com/hivetech/dna/blob/develop/LICENSE)

### CODE REVIEW

**Comment by Alhadis** [comment]
:wave: Hi @wesinator, thanks for your submission. This looks good; I corrected the search-result URL for you (and narrowed the results down to only include `bash_aliases` which don't contain hashbangs). Please be sure to link correctly next time.

I harvested some [~1062 files](https://github.com/Alhadis/Silos/blob/bash_aliases/urls.log) (out of ~16,198 total results), most of which are unique per user. Distribution most certainly checks out.

/cc @pchaigno, @lildude

**Comment by pchaigno** [comment]
Thanks a lot @Alhadis for running Harvester!

@wesinator Could you add a sample file as per the contribution guidelines? We prefer sample files from "real" applications (as they are used for tests and to train the Bayesian classifier) and under permissive licenses.

**Comment by Alhadis** [comment]
> Thanks a lot @Alhadis for running Harvester!

More like thanks [for *fixing*](https://github.com/Alhadis/Harvester/commit/2485145d8e9a75454227ab261448159aa75d07e8) Harvester. 😉

FYI, the dotted variant of the filename (`.bash_aliases`) commands considerable usage as well: I harvested [`1,850`](https://github.com/Alhadis/Silos/blob/bash_aliases/.urls.log) out of [~16,198 results](https://github.com/search?q=filename%3A.bash_aliases+-language%3Ashell+NOT+nothack&type=Code), which is much more than the undotted variant. We should include it as well.

File: github-linguist.gemspec [hunk 1]
```diff
@@ -29,4 +29,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'color-proximity', '~> 0.2.1'
   s.add_development_dependency 'licensed', '~> 1.3.0'
   s.add_development_dependency 'licensee'
+  s.add_development_dependency 'bundler', '~> 1.10'
```
Comment (pchaigno) [inline github-linguist.gemspec]: @wesinator You've included several changes by mistake I believe; these shouldn't appear.

**Comment by pchaigno** [comment]
Should we also add `bash_aliases` as a filename?

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -4564,6 +4564,7 @@ Shell:
   - ".tool"
   - ".zsh"
   filenames:
+  - ".bash_aliases"
```
Comment (Alhadis) [inline lib/linguist/languages.yml]: Could you please also add an entry for this without the dot?

~~~diff
   - 9fs
   - PKGBUILD
+  - bash_aliases
   - bash_logout
   - bash_profile
~~~

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -4578,6 +4579,7 @@ Shell:
   - ".zshrc"
   - 9fs
   - PKGBUILD
+  - bash_aliases
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Could you add a sample file for `bash_aliases` as well please?

**Pull request merged**